jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -136,6 +138,214 @@ ecma_builtin_typedarray_prototype_length_getter (ecma_value_t this_arg) /**< thi
 } /* ecma_builtin_typedarray_prototype_length_getter */
 
 /**
+ * Type of routine.
+ */
+typedef enum
+{
+  TYPEDARRAY_ROUTINE_EVERY, /**< routine: every */
+  TYPEDARRAY_ROUTINE_SOME, /**< routine: some */
+  TYPEDARRAY_ROUTINE_FOREACH, /**< routine: forEach */
+  TYPEDARRAY_ROUTINE__COUNT /**< count of the modes */
+} typedarray_routine_mode;
+
+/**
+ * The common function for 'every', 'some' and 'forEach'
+ * because they have a similar structure.
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_exec_routine (ecma_value_t this_arg, /**< this argument */
+                                                ecma_value_t arg1, /**< callbackfn */
```

Comment:
ok, i will change the var name.

no, arg2 is the 'this' for the callback function